tools/perfscale-load-generator/burst/burst.go

```diff
@@ -44,104 +44,115 @@ type BurstJob struct {
 		if err != nil {
 			continue
 		}
-		if b.objType == "" {
-			b.objType = objUtil.GetObjectResource(newObject)
-		}
+	}
 
-		wg.Add(1)
-		go func(newObject *unstructured.Unstructured) {
-			defer wg.Done()
-			b.watchCreate(newObject)
-			log.Log.Infof("obj %s is available", newObject.GroupVersionKind().Kind)
-		}(newObject)
+	// Wait all objects be Running
+	for objType, objWatcher := range b.watchers {
+		log.Log.Infof("Wait for obj(s) %s to be available", objType)
+		objWatcher.WaitRunning(b.Workload.Timeout.Duration)
 	}
-	wg.Wait()
 }
 
 func (b *BurstJob) DeleteWorkload() {
 	log.Log.V(1).Infof("Burst Load Generator DeleteWorkload")
-
-	obj := b.Workload.Object
-	getObject, objType := objUtil.FindObject(b.virtClient, obj, b.Workload.Count)
-	b.objType = objType
-	if getObject != nil {
-		labels := getObject.GetLabels()
-		jobUUID := labels[config.WorkloadLabel]
-		log.Log.V(2).Infof("Deleting all workloads for job %s", jobUUID)
-		objUtil.DeleteAllObjectsInNamespaces(b.virtClient, objType, config.GetListOpts(config.WorkloadLabel, jobUUID))
-		b.watchDelete(getObject)
+	objSpec := b.Workload.Object
+	objSample := renderObjSpecTemplate(objSpec, b.UUID)
+	b.createWatcherIfNotExist(objSample)
+
+	// Delete objects that match labels up to the count
+	objType := objUtil.GetObjectResource(objSample)
+	labels := objSample.GetLabels()
+	jobUUID := labels[config.WorkloadLabel]
+	log.Log.V(2).Infof("Deleting %d objects for job %s", b.Workload.Count, jobUUID)
+	objUtil.DeleteNObjectsInNamespaces(b.virtClient, objType, config.GetListOpts(config.WorkloadLabel, jobUUID), b.Workload.Count)
+
+	// Wait all objects be Deleted. In the case of VMI, deleted means the succeded phase.
+	for objType, objWatcher := range b.watchers {
+		log.Log.Infof("Wait for obj(s) %s to be deleted", objType)
+		objWatcher.WaitDeletion(b.Workload.Timeout.Duration)
 	}
-	log.Log.V(2).Infof("All workloads for job have been deleted")
-}
 
-func (b *BurstJob) watchDelete(obj *unstructured.Unstructured) {
-	objWatcher := watcher.NewObjListWatcher(
-		b.virtClient,
-		b.objType,
-		b.Workload.Count,
-		*config.GetListOpts(config.WorkloadLabel, b.UUID))
-	objWatcher.Run()
-
-	log.Log.Infof("Wait for obj(s) %s to be deleted", b.objType)
-	objWatcher.WaitDeletion(b.Workload.Timeout.Duration)
-	objWatcher.Stop()
+	log.Log.V(2).Infof("All workloads for job have been deleted")
 }
 
-func (b *BurstJob) watchCreate(obj *unstructured.Unstructured) {
-	objWatcher := watcher.NewObjListWatcher(
-		b.virtClient,
-		b.objType,
-		b.Workload.Count,
-		*config.GetListOpts(config.WorkloadLabel, b.UUID))
-	objWatcher.Run()
-
-	log.Log.Infof("Wait for obj(s) %s to be available", b.objType)
-	objWatcher.WaitRunning(b.Workload.Timeout.Duration)
-	objWatcher.Stop()
+func (b *BurstJob) createWatcherIfNotExist(obj *unstructured.Unstructured) {
+	objType := objUtil.GetObjectResource(obj)
+	if _, exist := b.watchers[objType]; !exist {
```

Comment:
nit: this is not thread safe, but we do not need to guard for parallelization here, right?

Reply:
We do not run parallel tests, so there is only one thread accessing the `b.watchers[objType]`